pkg/store/gate.go

```diff
@@ -0,0 +1,43 @@
+package store
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/gate"
+)
+
+// Gate wraps the Prometheus gate with extra metrics.
+type Gate struct {
+	g              *gate.Gate
+	currentQueries prometheus.Gauge
+}
+
+// NewGate returns a new gate.
+func NewGate(maxConcurrent int, reg prometheus.Registerer) *Gate {
+	g := &Gate{
+		g: gate.New(maxConcurrent),
+	}
+	g.currentQueries = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "thanos_bucket_store_queries_total",
```

Comment:
Gauge name should not end with the `_total` suffix

Reply:
Also might be useful to expose also `thanos_bucket_store_queries_limit`. This way you could add to your graph also the limit so user can see it he is hitting it without need to look to the configuration.

Reply:
Yup :+1: